### PR TITLE
Fix kk_KZ\Person::individualIdentificationNumber generation

### LINKED ISSUE #412: kk_KZ\Person::individualIdentificationNumber  incorect digit for century
### Summary

kk_KZ\Person::individualIdentificationNumber sometimes generates incorect digit for century.

### Versions

|                | Version |
|:---------------|:--------|
| PHP            | Php version independent   |
| `fakerphp/faker`  | main branch  |

### Self-enclosed code snippet for reproduction

```php
        $birthDate = DateTime::dateTimeBetween('2030-01-01', '2099-12-31');
        $individualIdentificationNumber = $this->faker->individualIdentificationNumber($birthDate, Person::GENDER_MALE);
```

### Expected output

```txt
        (int)$individualIdentificationNumber[6] == Person::MALE_CENTURY_21ST;
```

### Actual output

```txt
        (int)$individualIdentificationNumber[6] == Person::MALE_CENTURY_20ST
```

### PR DESCRIPTION
**Pull request by vadimonus**
### What is the reason for this PR?

Fix incorrect generation of 7 digit in kz individualIdentificationNumber

- [x] Fixed an issue (resolve #412)

### Author's checklist

- [x] Follow the [Contribution Guide](https://github.com/FakerPHP/Faker/blob/main/.github/CONTRIBUTING.md)
- [x] Bug fix, nothing to document

### Summary of changes

Fix incorrect generation of 7 digit in kz individualIdentificationNumber

### Review checklist

- [ ] All checks have passed
- [ ] Changes are approved by maintainer

### CODE REVIEW

File: src/Faker/Provider/kk_KZ/Person.php [hunk 1]
```diff
@@ -175,26 +175,21 @@ class Person extends \Faker\Provider\Person
         'Ісмет',
     ];
 
-    /**
-     * @param int $year
-     *
-     * @return int|null
-     */
-    private static function getCenturyByYear($year)
+    private static function getCenturyByYear(int $year): int
```
Comment (pimjansen) [inline src/Faker/Provider/kk_KZ/Person.php:178]: This is a BC and not allowed

File: test/Faker/Provider/kk_KZ/PersonTest.php [hunk 2]
```diff
@@ -13,11 +13,39 @@ final class PersonTest extends TestCase
 {
     public function testIndividualIdentificationNumberIsValid()
     {
-        $birthDate = DateTime::dateTimeBetween('-30 years', '-10 years');
-        $individualIdentificationNumber = $this->faker->individualIdentificationNumber($birthDate);
-        $controlDigit = Person::checkSum($individualIdentificationNumber);
+        // When calculating individual identification number,
```
Comment (pimjansen) [inline test/Faker/Provider/kk_KZ/PersonTest.php:16]: Please put this as example in the phpdoc

**Review by pimjansen — CHANGES_REQUESTED** [review]

File: src/Faker/Provider/kk_KZ/Person.php (hunk 1 above)
Comment (vadimonus) [reply src/Faker/Provider/kk_KZ/Person.php:178]: Please clarify, that is private method, so it cannot affect any descendant classes

Comment (pimjansen) [reply src/Faker/Provider/kk_KZ/Person.php]: Ah sorry you are ofc right

**Review by pimjansen — APPROVED** [review]
Lgtm

**Pull request merged**